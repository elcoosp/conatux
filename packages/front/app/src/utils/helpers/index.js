export const pipe = {
  execAsync: (...funcs) => context =>
    funcs.reduce(async (ctx, func) => {
      await func(ctx)
      return ctx
    }, context)
}
