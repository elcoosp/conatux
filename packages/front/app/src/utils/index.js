export { default as testing } from './testing'
export { default as linkify } from './linkify'
