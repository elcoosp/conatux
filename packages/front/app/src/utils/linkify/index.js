import React from 'react'
import { Link } from 'react-router-dom'

export const labelize = (path, label) => ({
  path,
  label
})

export const fromRoutes = (routesTree, componentName) => {
  if (['label', 'path'].every(prop => routesTree.hasOwnProperty(prop))) {
    const { path, label } = routesTree
    const link = props => (
      <Link to={path} {...props}>
        {label}
      </Link>
    )
    link.displayName = componentName
    return link
  }

  return Object.entries(routesTree).reduce(
    (links, [routeName, nestedRoute]) => {
      const componentName =
        routeName.slice(0, 1).toUpperCase() + routeName.slice(1)
      links[componentName] = fromRoutes(nestedRoute, componentName)
      return links
    },
    {}
  )
}
