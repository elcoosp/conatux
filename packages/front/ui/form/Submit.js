import React from 'react'
import P from 'prop-types'

const Submit = ({ disabled, ...props }) => (
  <button type="submit" disabled={disabled} {...props}>
    Submit
  </button>
)

Submit.propTypes = {
  disabled: P.bool.isRequired
}

export default Submit
