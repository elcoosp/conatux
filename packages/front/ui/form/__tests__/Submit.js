import { desc, getFakeProps } from '@:/utils/testing'
import Submit from '../Submit'

const [props] = getFakeProps(__dirname, '../Submit')('Submit')

desc.compWithSnap({ Submit }, props, () => {})
