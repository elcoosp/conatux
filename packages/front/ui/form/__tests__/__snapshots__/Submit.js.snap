// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Submit /> should match snapshot 1`] = `
<button
  disabled={true}
  type="submit"
>
  Submit
</button>
`;
